### PR TITLE
support gcloud spanner in tsh and web UI

### PR DESCRIPTION
**Pull request by GavinFrazar**
This was extracted from and builds on the spanner engine PR:
- https://github.com/gravitational/teleport/pull/40859

It adds tsh support and web ui audit event support.

Will be rebased automatically onto master after the engine PR merges.

A docs update will be coming in the next PR.

### CODE REVIEW

File: web/packages/teleport/src/services/audit/makeEvent.ts [hunk 1]
```diff
@@ -1823,6 +1823,32 @@ export const formatters: Formatters = {
     format: ({ user }) =>
       `User [${user}] updated the cluster session recording configuration`,
   },
+  [eventCodes.SPANNER_RPC]: {
+    type: 'db.session.spanner.rpc',
+    desc: 'Spanner RPC',
+    format: ({ args, user, procedure, db_name, db_service }) => {
+      if (args.sql) {
+        return `User [${user}] executed query [${truncateStr(
```
Comment (greedy52) [inline web/packages/teleport/src/services/audit/makeEvent.ts:1831]: Thanks! 🙏

File: tool/tsh/common/proxy.go [hunk 2]
```diff
@@ -706,6 +748,19 @@ Use the following command to connect to the database or to the address above usi
...
+var dbProxySpannerAuthTpl = template.Must(template.New("").Parse(
+	`Started authenticated tunnel for the {{.type}} database "{{.database}}" in cluster "{{.cluster}}" on {{.address}}.
+{{if .randomPort}}To avoid port randomization, you can choose the listening port using the --port flag.
+{{end}}
+` + dbProxyConnectAd + `
+Use the following command to connect to the database or to the address above using other database GUI/CLI clients:
+  $ {{.command}}
+
+Or use a JDBC connection string to connect with other GUI/CLI clients:
+jdbc:cloudspanner://{{.address}}/projects/{{.gcpProject}}/instances/{{.gcpInstance}}/databases/{{.databaseName}}
```
Comment (greedy52) [inline tool/tsh/common/proxy.go:761]: could we add the `usePlainText` option here?

**Review by greedy52 — APPROVED** [review]

File: tool/tsh/common/proxy.go (hunk 2 above)
Comment (GavinFrazar) [reply tool/tsh/common/proxy.go]: [1dec6b0](https://github.com/gravitational/teleport/pull/41129/commits/1dec6b00fc0329a370aa4a3bcd1d85648070ed6a)

**Comment by GavinFrazar** [comment]
I think the "buf breaking" lint error is just because this PR is currently based on my other feature branch instead of master, and since I force-pushed that branch it thinks there's been a buf breaking change. It should clear after the base branch merges to master.

**Review by smallinsky — APPROVED** [review]

**Review by Tener — APPROVED** [review]

**Pull request merged**